src/utils/weights.rs

```diff
@@ -63,13 +67,33 @@ impl WeightInit {
         Value::from(weight)
     }
 
-    fn he_normal(&self, fan_num: usize) -> Value {
-        let stdev = 1.0 / (fan_num as f64).sqrt();
+    fn he_normal(&self, fan_in: usize) -> Value {
+        let stdev = (2.0 / fan_in as f64).sqrt();
         let normal = Normal::new(0.0, stdev).unwrap();
 
         let mut rng = rand::thread_rng();
 
         let weight = normal.sample(&mut rng);
         Value::from(weight)
     }
+
+    fn lecun_uniform(&self, fan_in: usize) -> Value{
+        let limit = (3.0 / fan_in as f64).sqrt();
+        let uniform = Uniform::new(-limit, limit);
+
+        let mut rng = rand::thread_rng();
+
+        let weight = uniform.sample(&mut rng);
+        Value::from(weight)
+    }
+
+    fn lecun_normal(&self, fan_in: usize) -> Value{
```

Comment:
This might fail fmt test since there is no spacing between `Value` and `{`. Try running `cargo fmt` and `cargo clippy` if the CI tests fail

Reply:
alright will work on a fix